### PR TITLE
chore: update error messages

### LINKED ISSUE #843: update error messages to provide helpful, readable information
### Describe what should be investigated or refactored
We have noticed that where we throw an error and only use `e` as the message, we may not display readable and helpful information in the Pepr logs. Throwing the error with a specific message results in the Pepr logs displaying helpful information about what went wrong. We should check Pepr and KFC for additional instances of throwing errors using just `e` and update the error messages.

### Links to any relevant code
(optional) i.e. - https://github.com/defenseunicorns/pepr/blob/24f1f253b1db4c9d97717c0282b8f39c74d3ace1/README.md?plain=1#L3

### Additional context
Add any other context or screenshots about the technical debt here.

### PR DESCRIPTION
**Pull request by schaeferka**
## Description
We have noticed that where we throw an error and only use e as the message, we may not display readable and helpful information in the Pepr logs. Throwing the error with a specific message results in the Pepr logs displaying helpful information about what went wrong. We should check Pepr and KFC for additional instances of throwing errors using just e and update the error messages.
...

## Related Issue

Fixes #843 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/cli/build.ts [hunk 1]
```diff
@@ -310,7 +310,7 @@ export async function buildModule(reloader?: Reloader, entryPoint = peprTS, embe
 
     return { ctx, path, cfg, uuid };
   } catch (e) {
-    console.error(e.message);
+    console.error(`Error building module: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/build.ts]:
```suggestion
    console.error(e, `Error building module`);
```

File: src/cli/dev.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export default function (program: RootCmd) {
             // wait for capabilities to be loaded and test names
             validateCapabilityNames(webhook.capabilities);
           } catch (e) {
-            console.error(e.message);
+            console.error(`Error validating capability names: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/dev.ts]:
```suggestion
            console.error(e, `Error validating capability names`);
```

File: src/cli/format.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ export async function peprFormat(validateOnly: boolean) {
 
       return !hasFailure;
     } catch (e) {
-      console.error(e.message);
+      console.error(`Error formatting module: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/format.ts]:
```suggestion
      console.error(e, `Error formatting module`);
```

File: src/cli/init/index.ts [hunk 4]
```diff
@@ -90,7 +90,7 @@ export default function (program: RootCmd) {
           console.log(`Open VSCode or your editor of choice in ${dirName} to get started!`);
         } catch (e) {
           if (e instanceof Error) {
-            console.error(e.message);
+            console.error(`Error creating Pepr module: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/init/index.ts]:
```suggestion
            console.error(e, `Error creating Pepr module`);
```

File: src/cli/kfc.ts [hunk 5]
```diff
@@ -38,7 +38,7 @@ export default function (program: RootCmd) {
           stdio: "inherit",
         });
       } catch (e) {
-        console.error(e.message);
+        console.error(`Error creating CRD generated class: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/kfc.ts]:
```suggestion
        console.error(e, `Error creating CRD generated class`);
```

File: src/cli/update.ts [hunk 6]
```diff
@@ -55,7 +55,7 @@ export default function (program: RootCmd) {
 
         console.log(`✅ Module updated successfully`);
       } catch (e) {
-        console.error(e.message);
+        console.error(`Error updating Pepr module: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/update.ts]:
```suggestion
        console.error(e, `Error updating Pepr module`);
```

File: src/cli/update.ts [hunk 7]
```diff
@@ -88,7 +88,7 @@ export default function (program: RootCmd) {
           }
         }
       } catch (e) {
-        console.error(e.message);
+        console.error(`Error updating template files: ${e.message}`);
```
Comment (cmwylie19) [inline src/cli/update.ts]:
```suggestion
        console.error(e, `Error updating template files`);
```

**Comment by cmwylie19** [comment]
my two cents are that we should dump the whole error in the output

File: src/cli/build.ts (hunk 1 above)
Comment (schaeferka) [reply src/cli/build.ts]: All messages updated to display complete `e`

File: src/cli/dev.ts (hunk 2 above)
Comment (schaeferka) [reply src/cli/dev.ts]: All messages updated to display complete `e`

File: src/cli/format.ts (hunk 3 above)
Comment (schaeferka) [reply src/cli/format.ts]: All messages updated to display complete `e`

File: src/cli/init/index.ts (hunk 4 above)
Comment (schaeferka) [reply src/cli/init/index.ts]: All messages updated to display complete `e`

File: src/cli/kfc.ts (hunk 5 above)
Comment (schaeferka) [reply src/cli/kfc.ts]: All messages updated to display complete `e`

File: src/cli/update.ts (hunk 6 above)
Comment (schaeferka) [reply src/cli/update.ts]: All messages updated to display complete `e`

File: src/cli/update.ts (hunk 7 above)
Comment (schaeferka) [reply src/cli/update.ts]: All messages updated to display complete `e`

File: src/lib/controller/index.ts [hunk 8]
```diff
@@ -182,7 +182,7 @@ export class Controller {
     try {
       res.send(await this.#metricsCollector.getMetrics());
     } catch (err) {
-      Log.error(err);
+      Log.error(err, `Error getting metrics: ${err}`);
```
Comment (cmwylie19) [inline src/lib/controller/index.ts]:
```suggestion
      Log.error(err, "Error getting metrics");
```

Comment (schaeferka) [reply src/lib/controller/index.ts]: Updates made

**Review by cmwylie19 — APPROVED** [review]

**Pull request merged**